src/Domain/BankDataValidator.php

```diff
@@ -4,69 +4,79 @@
 
 namespace WMDE\Fundraising\PaymentContext\Domain;
 
+use WMDE\Fundraising\PaymentContext\Domain\BankDataValidationResult as Result;
 use WMDE\Fundraising\PaymentContext\Domain\Model\BankData;
-use WMDE\FunValidators\CanValidateField;
 use WMDE\FunValidators\ConstraintViolation;
 use WMDE\FunValidators\ValidationResult;
-use WMDE\FunValidators\Validators\RequiredFieldValidator;
-use WMDE\FunValidators\Validators\StringLengthValidator;
 
 /**
  * @license GNU GPL v2+
  * @author Kai Nissen < kai.nissen@wikimedia.de >
  */
 class BankDataValidator {
-	use CanValidateField;
 
-	public const INVALID_BIC = 'invalid_bic';
-
-	private const BANK_ACCOUNT_MAXLEN = 10;
-	private const BANK_CODE_MAXLEN = 8;
-	private const BANK_NAME_MAXLEN = 250;
+	private $maximumFieldLengths = [
+		Result::SOURCE_BANK_ACCOUNT => 10,
+		Result::SOURCE_BANK_CODE => 8,
+		Result::SOURCE_BANK_NAME => 250,
+	];
 
 	private $ibanValidator;
 
+	private $violations;
+
 	public function __construct( IbanValidator $ibanValidator ) {
 		$this->ibanValidator = $ibanValidator;
 	}
 
 	public function validate( BankData $bankData ): ValidationResult {
-		$requiredValidator = new RequiredFieldValidator();
-		$stringLengthValidator = new StringLengthValidator();
-		$violations = [];
+		$this->violations = [];
```

Comment:
I know you did not introduce this here though want to point out this method is rather loooooooooooooooooooongcatislooooooooooooooooooong

Reply:
Thanks for the comment, I've extracted a method and now the `validate` method is more readable.